src/bundle-manifest.ts

```diff
@@ -21,15 +22,54 @@ export type UrlString = string;
  * entrypoint files.
  */
 export class Bundle {
-  public entrypoints: Set<UrlString>;
-  public files: Set<UrlString>;
+  entrypoints: Set<UrlString>;
+  files: Set<UrlString>;
+
   constructor(entrypoints?: Set<UrlString>, files?: Set<UrlString>) {
     this.entrypoints = entrypoints || new Set<UrlString>();
     this.files = files || new Set<UrlString>();
   }
 }
 
 /**
+ * A bundle manifest is a mapping of urls to bundles.
+ */
+export class BundleManifest {
+  // Map of bundle url to bundle.
+  bundles: Map<UrlString, Bundle>;
+
+  // Map of file url to bundle url.
+  files: Map<UrlString, UrlString>;
```

Comment:
The name `files` with the same type for key and value could be a little vague. A name like `bundleUrlForFile` might read better at the use site: `bundleManifest.bundleUrlForFile.get(url)`.

The other option is to make this private and expose a method: `bundleManifest.getBundleUrlForFile(url)`